### PR TITLE
Improve the version listview

### LINKED ISSUE #6070: Version's page is slow
Currently, we list all versions in one page. This is a problem for projects that have thousands of versions (it's difficult to search the versions and very slow for the db)

https://readthedocs.org/projects/docs/versions/

One idea is to don't list deactivated versions and use a form instead. Not really sure how that looks like, we still need a way to list all versions. If we put an auto complete widget, does the user know what version wants to activate?

Ref https://github.com/readthedocs/readthedocs.org/issues/6068#issuecomment-520908761

### PR DESCRIPTION
**Pull request by davidfischer**
Update the version listview screen with the following:

- No more than 100 inactive versions
- Filter inactive versions by name
- Fixes #6070

## Screenshot
Screen Shot 2019-09-30 at 11 48 11 PM

### CODE REVIEW

**Comment by davidfischer** [comment]
I should clarify that the whole "Activate a Version" section only displays for project maintainers just like the previous "Inactive Versions" section it replaces.

File: readthedocs/templates/projects/project_version_list.html [hunk 1]
```diff
@@ -104,7 +119,7 @@ <h1>{% trans "Inactive Versions" %}</h1>
 
                       {% if request.user|is_admin:project %}
                         <ul class="module-item-menu">
-                          <li><a href="{% url "project_version_detail" project.slug version.slug %}">{% trans "Edit" %}</a></li>
+                          <li><a href="{% url "project_version_detail" project.slug version.slug %}">{% trans "Edit & Activate" %}</a></li>
```
Comment (humitos) [inline readthedocs/templates/projects/project_version_list.html]: I don't like the "Edit & Activate" label for the button, but I do not have a better one. Do you think that just "Edit" as it was does not communicate it properly?

Comment (agjohnson) [reply readthedocs/templates/projects/project_version_list.html]: I might have the same feeling. The phrase sounds overly specific for a user. Technically, from our perspective, they are editing and activating, but to the user, they are just activating. I think "Activate" is the most correct descriptor for what the user is looking to accomplish, we can gloss over the fact that it drops into a form.

Comment (davidfischer) [reply readthedocs/templates/projects/project_version_list.html]: Sounds good. I'll update it.

**Comment by davidfischer** [comment]
I changed the text on that button "Edit & Activate" -> "Activate".

This should be merge ready. For the fix to take effect on RTD for Business where that template is overridden, I will create a corresponding private PR.

**Pull request merged**